compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs

```diff
@@ -216,9 +236,40 @@ loadPackages importPkgs replClient ideState = do
       , mod <- NM.names $ LF.packageModules pkg
       ]
 
+data ReplLogger = ReplLogger
+  { withReplLogger :: forall a. ([FileDiagnostic] -> IO ()) -> IO a -> IO a
+  -- ^ Temporarily modify what happens to diagnostics
+  , replEventLogger :: FromServerMessage -> IO ()
+  -- ^ Logger to pass to `withDamlIdeState`
+  }
+
 
-runRepl :: [(LF.PackageName, Maybe LF.PackageVersion)] -> Options -> ReplClient.Handle -> IdeState -> IO ()
-runRepl importPkgs opts replClient ideState = do
+getReplLogger :: IO ReplLogger
```

Comment:
Maybe a better name would be `newReplLogger` or `createReplLogger`? `getReplLogger` sounds like it would return a reference to a shared object.

Reply:
Good call, renamed to `newReplLogger`.